packages/react-components/src/helpers/format-date.js

```diff
@@ -0,0 +1,55 @@
+export const convertToDate = date => {
+  if (!date || !(date instanceof Date || typeof date === 'number')) return;
+  return new Date(date);
+};
```

Comment:
Would we need this if we always used `Date` objects? Maybe a better question is - can/should we ensure that we are always using `Date` objects (without this function) as opposed to numbers?

Reply:
I think we should continue to convert to a new Date object because it removes the risk of mutating the original Date object when using native methods such as `setHours`. `typeof date === 'number'` was added to allow some flexibility in case `Date.now()` was used, but we can replace it with `new Date()`.

Reply:
If we're only using `setHours` in the `isDateSameDay` function, could we avoid the risk of modifying the original with something like this?

```js
return (
  date1.getFullYear() === date2.getFullYear() &&
  date1.getMonth() === date2.getMonth() &&
  date1.getDay() === date2.getDay());
```

Reply:
That works beautifully. Updated to remove creation of new Date objects and `isDateSameDay` with this suggestion.